src/WinRT.Runtime/Marshalers.cs

```diff
@@ -740,33 +779,52 @@ internal sealed class MarshalGenericFallback<T>
         private readonly MethodInfo _fromManagedArray;
         private readonly MethodInfo _disposeMarshalerArray;
         private readonly MethodInfo _disposeAbiArray;
+#endif
 
         public MarshalGenericFallback(
 #if NET
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
 #endif
             Type helperType)
         {
-            _createMarshaler = helperType.GetMethod("CreateMarshaler", BindingFlags.Public | BindingFlags.Static);
-            _getAbi = helperType.GetMethod("GetAbi", BindingFlags.Public | BindingFlags.Static);
-            _copyAbi = helperType.GetMethod("CopyAbi", BindingFlags.Public | BindingFlags.Static);
-            _fromAbi = helperType.GetMethod("FromAbi", BindingFlags.Public | BindingFlags.Static);
-            _fromManaged = helperType.GetMethod("FromManaged", BindingFlags.Public | BindingFlags.Static);
-            _copyManaged = helperType.GetMethod("CopyManaged", BindingFlags.Public | BindingFlags.Static);
-            _disposeMarshaler = helperType.GetMethod("DisposeMarshaler", BindingFlags.Public | BindingFlags.Static);
-            _createMarshaler2 = helperType.GetMethod("CreateMarshaler2", BindingFlags.Public | BindingFlags.Static);
-            _disposeAbi = helperType.GetMethod("DisposeAbi", BindingFlags.Public | BindingFlags.Static);
-            _createMarshalerArray = helperType.GetMethod("CreateMarshalerArray", BindingFlags.Public | BindingFlags.Static);
-            _getAbiArray = helperType.GetMethod("GetAbiArray", BindingFlags.Public | BindingFlags.Static);
-            _fromAbiArray = helperType.GetMethod("FromAbiArray", BindingFlags.Public | BindingFlags.Static);
-            _fromManagedArray = helperType.GetMethod("FromManagedArray", BindingFlags.Public | BindingFlags.Static);
-            _disposeMarshalerArray = helperType.GetMethod("DisposeMarshalerArray", BindingFlags.Public | BindingFlags.Static);
-            _disposeAbiArray = helperType.GetMethod("DisposeAbiArray", BindingFlags.Public | BindingFlags.Static);
+#if NET8_0_OR_GREATER
+            _createMarshaler = MarshalExtensions.TryGetMethodInvoker(helperType, "CreateMarshaler");

```

Comment:
@MichalStrehovsky additional question: is creating a `MethodInvoker` instance fast enough that in this scenario (ie. we're already doing some reflection lookups) it doesn't matter and we continue eagerly initializing these instances, or is it recommended to load them on the fly with some kind of lazy loading strategy? Ie. is this current code ok here?

Reply:
No need for lazy loading.